app/models/communication.rb

```diff
@@ -60,6 +61,20 @@ def self.create_with_twilio_details!(appointment:, twilio_sid:, twilio_msg_statu
     end
   end
 
+  def self.create_with_imo_details!(appointment:, notification:, result:)
+    transaction do
+      # create detailable here once detailable model exists
+      communication = Communication.create!(communication_type: "imo",
+                                            detailable: nil,
+                                            appointment: appointment,
+                                            notification: notification,
+                                            device_created_at: DateTime.current,
+                                            device_updated_at: DateTime.current)
```

Comment:
Extract `DateTime.current` into a `now` var, instead of invoking twice? If we're trying to emulate a device creating a communication, then we should ensure the timestamps match exactly.

Reply:
Fair point. I'll correct it in the twilio variation too.